lib/src/modules/task.cc

```diff
@@ -128,11 +136,171 @@ static TaskCommand getTaskCommand(const fs::path &task_executable)
     return TaskCommand { task_executable.string(), { } };
 }
 
+// Converts boost filesystem errors to a task_error object.
+// TODO: Modify this to convert it to a PXP-error instead.
```

Comment:
I did that because executeAction will catch the exception and wrap it up in a PXP error: https://github.com/puppetlabs/pxp-agent/blob/master/lib/src/module.cc#L97. Should I include a separate catch block instead for filesystem errors?